ide/src/error_types.ts

```diff
@@ -0,0 +1,38 @@
+import vscode from "vscode";
+import { log, show_error } from "./vsc_utils";
+import { tdcp } from "./extension";
+
```

Comment:
I think a simpler design might be:

```ts
interface BuildError { type: "build-error", error: string }
interface AnalysisError { type: "analysis-error", error: string }
interface FlowistryOutput<T> { type: "output", value: T {
type FlowistryResult<T> = FlowistryOutput<T> | BuildError | AnalysisError

let show = (error: BuildError | AnalysisError) => {
  // ...
}

let is_ok = (res: FlowistryResult<T>): res is FlowistryOutput<T> => res.type == "output";
```

Then when you're using it:

```ts
let result = call_flowistry(...);
if (!is_ok(result)) {
  show(result);
  return;
}

process(result);
```

A few reasons:
* This avoids the `result as T` casting that's necessary otherwise.
* This reduces the boilerplate with error interfaces and class definitions. 
* We don't need errors that extend `Error` since we're not throwing exceptions.